src/components/DonorDetails/Edit.js

```diff
@@ -0,0 +1,168 @@
+          )
+        }
+      })
+      .catch((err) => console.log(err))
+  }
+  return (
+    <>
+      <Modal
+        show={props.showModal}
+        dialogClassName="modal-90w"
+        onHide={props.close}
+        className= "edit-modal-container"
+      >
+        
+        <div className="edit-modal">
+          <Modal.Header>
+            <header className="title">Edit Donor Information</header>
+            <Modal.Close onClick={props.close} />
+          </Modal.Header>
+
+          <Modal.Body className="edit-body">
+            <div className="edit-body-wrapper">
+              <div className="remarks-container">
+                <h1 className="edit-subheader">Remarks</h1>
+                <textarea
+                  name="Text1"
+                  id="remarks"
+                  value={remarks}
+                  className="remarks-input"
+                  onChange={(e) => setRemarks(e.target.value)}
+                ></textarea>
+              </div>
+              <div className="preference-container">
+                <h2 className="edit-subheader"> Preferred Contact </h2>
+                <div className="radio-container">
+                  <form onChange={handleChangedPreference}>
+                    {contacts.length &&
+                      contacts.map((contact) => (
+                        <div>
+                          <label>
+                            <input
+                              name="contact"
+                              type="radio"
+                              value={contact.id}
+                              defaultChecked={
+                                currentPreferredContact === contact.id
+                              }
+                            />
+                            
+                            <span className= "radio-label">{displayContactType(contact.id)}</span>
+                          </label>
+                        </div>
+                      ))}
+                    <label>
+                      <input
+                        name="contact"
+                        type="radio"
+                        value={true}
+                        defaultValue={currentDNC}
+                        defaultChecked={currentDNC}
+                      />
+                      <span className= "radio-label">Do Not Contact</span>
+                    </label>
+                  </form>
+                </div>
+              </div>
+            </div>
+          </Modal.Body>
+        </div>
+        <Modal.Footer>
+          <div className="edit-errormsg">{errorMsg}</div>
+          <button className="button transparentbutton" onClick={props.close}>
+            Cancel
+          </button>
+          <div>
+            <button className="button orangebutton" onClick={submitEdits}>
+              Save Donor Details
+            </button>
+            <br />
+          </div>
+        </Modal.Footer>
+      </Modal>
+    </>
+  )
+}
+
+const displayContactType = (id) => {
```

Comment:
Preferably, the backend should return the string along with the id. Currently it's almost there, in the `description` key:
![image](https://user-images.githubusercontent.com/3888250/79076822-8d8d3d00-7d2f-11ea-88c9-fe8a2de24286.png)